### PR TITLE
feat: Remove xformers, move to pytorch

### PR DESCRIPTION
**Pull request by sandhawalia**
xformers are notoriously difficult to port to onnx and this is a blocker for running rmind on the car. 

This PR:
- replicates the xformers architecture
- adds gradient checkpoint 
- sanity PT run ongoing 

One key difference w.r.t [xformers](https://github.com/facebookresearch/xformers/blob/v0.0.28.post2/xformers/components/reversible.py#L80) how the residual path is constructed for the mlp in the Encoder Block. This PR follows the industry standard of 

# this pr
```
x = x + mha(norm(x)
x = x + mlp(norm(x))
```

vs 

# xformers
```
x1, x2 = x, x
y1 = x1 + mha(norm(x1)
y2 = x2 + mlp(norm(y1))

```

### CODE REVIEW

File: config/datamodule/yaak/train.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ defaults:
 _target_: rmind.datamodules.GenericDataModule
 train:
   _target_: torch.utils.data.DataLoader
-  batch_size: 64
+  batch_size: 48
```
Comment (sandhawalia) [inline config/datamodule/yaak/train.yaml:9]: Still experimenting whats the largest I can push

File: config/model/yaak/control_transformer/raw.yaml [hunk 2]
```diff
@@ -15,9 +15,9 @@ input_builder:
       brake_pedal_diff: [data, meta/VehicleMotion/brake_pedal_normalized]
       steering_angle: [data, meta/VehicleMotion/steering_angle_normalized]
       steering_angle_diff: [data, meta/VehicleMotion/steering_angle_normalized]
-      
+
```
Comment (sandhawalia) [inline config/model/yaak/control_transformer/raw.yaml:18]: ?

**Comment by sandhawalia** [comment]
https://wandb.ai/yaak/cargpt/runs/589krxdk

File: pyproject.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [project]
 name = "rmind"
-version = "0.1.0"
+version = "0.2.0"
```
Comment (egorchakov) [inline pyproject.toml:3]: we dont publish this so version is somewhat meaningless; keep for now?

File: src/rmind/components/llm.py [hunk 4]
```diff
@@ -1,46 +1,116 @@
...
+        embedding_dim: int,
+        num_heads: int,
+        attn_dropout: float = 0.1,
+        resid_dropout: float = 0.1,
+        mlp_dropout: float = 0.1,
+        hidden_layer_multiplier: int = 1,
+    ) -> None:
+        super().__init__()
+
+        # f
```
Comment (egorchakov) [inline src/rmind/components/llm.py:19]: maybe more comment context on xformers equivalency?

File: src/rmind/components/llm.py [hunk 5]
```diff
@@ -1,46 +1,116 @@
...
+                resid_dropout=resid_dropout,
+                hidden_layer_multiplier=hidden_layer_multiplier,
+            )
+            for _ in range(num_layers)
+        ])
+        # https://github.com/karpathy/nanoGPT/blob/master/model.py#L182
+        self.layer_norm: nn.LayerNorm = nn.LayerNorm(dim_model)
+
+    @override
+    def forward(self, src: Tensor, mask: Tensor) -> Tensor:
```
Comment (egorchakov) [inline src/rmind/components/llm.py:82]: since they're both tensors, maybe kwargs to make arg order errors less likely at callsite?
```suggestion
    def forward(self, *, src: Tensor, mask: Tensor) -> Tensor:
```

File: src/rmind/components/llm.py [hunk 6]
```diff
@@ -1,46 +1,116 @@
...
+        x = src
+
+        for layer in self.layers:
+            if self.training:
```
Comment (egorchakov) [inline src/rmind/components/llm.py:86]: [nit] rather than checking `self.training` every time (presumably doesn't change as we iterate over layers), lift the if-guard up and check once?

File: src/rmind/components/llm.py [hunk 7]
```diff
@@ -1,46 +1,116 @@
...
-        self.d1 = nn.Dropout(dropout)
-        self.l2 = nn.Linear(in_features=dim_mlp, out_features=dim_model, bias=bias)
-        self.d2 = nn.Dropout(dropout)
+        self.l1: nn.Linear = nn.Linear(
+            in_features=dim_model, out_features=dim_mlp * 2, bias=bias
+        )
+        self.a1: nn.GELU = nn.GELU()
+        self.d1: nn.Dropout = nn.Dropout(dropout)
+        self.l2: nn.Linear = nn.Linear(
+            in_features=dim_mlp, out_features=dim_model, bias=bias
```
Comment (egorchakov) [inline src/rmind/components/llm.py:112]: [nit] imo annotations don't add much value here, can wrap in `@final` if ruff complains

**Review by egorchakov — CHANGES_REQUESTED** [review]

File: src/rmind/components/llm.py (hunk 6 above)
Comment (egorchakov) [reply src/rmind/components/llm.py:90]: how so?

Comment (sandhawalia) [reply src/rmind/components/llm.py:90]: you are right it doesn't
